spec/schema_classes/a_record.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# This file is autogenerated by Deimos, Do NOT modify
+module Deimos
```

Comment:
Just realized something... are these being generated as separate files? We might run into issues if there are multiple Avro schemas that have records defined in them and those records' names conflict. Probably better to generate them as inner classes. :(

Reply:
i _think_ i can do something similar and create a template for inner classes (Using something like `insert_into_file` or just plain ERB template loading from a helper method :p) - will probably shove all the inner classes _above_ the original logic

Reply:
Yep, something like that.

Reply:
Here's an example of what i've got so far :)

```ruby
# frozen_string_literal: true

# This file is autogenerated by Deimos, Do NOT modify
module Deimos
  # Primary Schema Class
  # Autogenerated Schema for Record at com.my-namespace.MyNestedSchema
  class MyNestedSchema < SchemaClass::Record
    # Attribute Readers
    # @return [Deimos::MyNestedRecord]
    attr_reader :some_nested_record
    # @return [nil, Deimos::MyNestedRecord]
    attr_reader :some_optional_record
    # Attribute Accessors
    # @param value [String]
    attr_accessor :test_id
    # @param value [Float]
    attr_accessor :test_float
    # @param values [Array<String>]
    attr_accessor :test_array
    # @return [Object] An optional payload key
    attr_accessor :payload_key
    # Attribute Writers
    # @param value [Deimos::MyNestedRecord]
    def some_nested_record=(value)
      @some_nested_record = Deimos::MyNestedRecord.initialize_from_value(value)
    end

    # @param value [nil, Deimos::MyNestedRecord]
    def some_optional_record=(value)
      @some_optional_record = Deimos::MyNestedRecord.initialize_from_value(value)
    end


    # @override
    def initialize(test_id: nil,
                   test_float: nil,
                   test_array: nil,
                   some_nested_record: nil,
                   some_optional_record: nil,
                   payload_key: nil)
      super()
      self.test_id = test_id
      self.test_float = test_float
      self.test_array = test_array
      self.some_nested_record = some_nested_record
      self.some_optional_record = some_optional_record
      self.payload_key = payload_key
    end

    # @override
    def schema
      'MyNestedSchema'
    end

    # @override
    def namespace
      'com.my-namespace'
    end

    # @override
    def to_h
      payload = {
        'test_id' => @test_id,
        'test_float' => @test_float,
        'test_array' => @test_array,
        'some_nested_record' => @some_nested_record&.to_h,
        'some_optional_record' => @some_optional_record&.to_h
      }
      @payload_key.present? ? payload.merge('payload_key' => @payload_key) : payload
    end

    # Secondary Schema Classes
      # Autogenerated Schema for Record at com.my-namespace.MyNestedRecord
    class MyNestedRecord < SchemaClass::Record
      # Attribute Accessors
      # @param value [Integer]
      attr_accessor :some_int
      # @param value [Float]
      attr_accessor :some_float
      # @param value [String]
      attr_accessor :some_string
      # @param value [nil, Integer]
      attr_accessor :some_optional_int
      # @return [Object] An optional payload key
      attr_accessor :payload_key

      # @override
      def initialize(some_int: nil,
                     some_float: nil,
                     some_string: nil,
                     some_optional_int: nil,
                     payload_key: nil)
        super()
        self.some_int = some_int
        self.some_float = some_float
        self.some_string = some_string
        self.some_optional_int = some_optional_int
        self.payload_key = payload_key
      end

      # @override
      def schema
        'MyNestedRecord'
      end

      # @override
      def namespace
        'com.my-namespace'
      end

      # @override
      def to_h
        payload = {
          'some_int' => @some_int,
          'some_float' => @some_float,
          'some_string' => @some_string,
          'some_optional_int' => @some_optional_int
        }
        @payload_key.present? ? payload.merge('payload_key' => @payload_key) : payload
      end
    end
  
  end
end
```

Reply:
Good start! I don't think I would nest the generated modules under `Deimos` because these are part of the app, not Deimos itself. Also, I think you'd need to put the nested definitions above the main one, because otherwise you'll get a crash when it tries to use them :)

Reply:
ooooo really - can move above, would it look something like this?

```ruby
# This file is autogenerated by Deimos, Do NOT modify

# Secondary Schema Classes
# Autogenerated Schema for Record at com.my-namespace.MyNestedRecord
class MyNestedRecord < Deimos::SchemaClass::Record
  ...
end

# :nodoc:
module Deimos
  # Primary Schema Class
  # Autogenerated Schema for Record at com.my-namespace.MyNestedSchema
  class MyNestedSchema < SchemaClass::Record
    ...
  end
end
```

Reply:
Pushed le change for this and some of the resolved ones above!